### PR TITLE
Address feedback + missing some tests

### PR DESCRIPTION
**Pull request by j0shuams**

### CODE REVIEW

**Comment by Scottj1s** [comment]
>         private const string UnsealedClass = @"namespace Test { public class UnsealedClass { public UnsealedClass() {} } }";

nit: since you're using C# verbatim strings, consider adding linefeeds for readability

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:21 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>    public sealed class OpWithProgress : IAsyncOperationWithProgress<int, bool>

nit: space

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:699 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
> public class ActionWithProgress : IAsyncActionWithProgress<int>

indent

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:731 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>         void IAsyncInfo.Cancel()

these can all use member-bodied expressions too

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:710 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>    public sealed class AsyAction : IAsyncAction

space

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:794 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>    public interface OpWithProgress : IAsyncOperationWithProgress<int, bool>

space

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:846 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>     {

some spacing issues

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:1200 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>         public sealed class SillyClass

indentation here and down

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:1211 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

**Comment by Scottj1s** [comment]
>         [Windows.Foundation.Metadata.Deprecated(""hu"", Windows.Foundation.Metadata.DeprecationType.Deprecate, 1), 

what does "hu" indicate?  if it's unused, maybe change to something less distracting?  (like "deprecated")

---
Refers to: src/Authoring/DiagnosticTests/NegativeData.cs:1361 in dc1070f. [](commit_id = dc1070f55a9a9d724ccb6f96f18778ed13d0cbce, deletion_comment = False)

File: src/Authoring/DiagnosticTests/PositiveData.cs [hunk 1]
```diff
@@ -1,802 +1,824 @@
...
+    public sealed Class1 { public int X { get; set; }  }
+
+    namespace InnerComponent
+    {
+        public sealed class Class2 { public int Y { get; } }
+    }
+}
+";
+        /**/
+        /* Need to check method parameters now, just like we */
```
Comment (Scottj1s) [inline src/Authoring/DiagnosticTests/PositiveData.cs:17]: comment?

File: src/Authoring/DiagnosticTests/UnitTestDiagnostics.cs [hunk 2]
```diff
@@ -148,6 +149,27 @@ private static IEnumerable<TestCaseData> InvalidCases
                 yield return new TestCaseData(SubNamespace_Jagged3DInterface6, DiagnosticRules.ArraySignature_JaggedArrayRule).SetName("Jagged 3D Array SubNamespace Interface Method 6");
 
                 // overload attribute tests
+               yield return new TestCaseData(TwoOverloads_NoAttribute_NamesHaveNumber, DiagnosticRules.MethodOverload_NeedDefaultAttribute)
```
Comment (Scottj1s) [inline src/Authoring/DiagnosticTests/UnitTestDiagnostics.cs:151]: space

File: src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs [hunk 3]
```diff
@@ -219,163 +224,108 @@ public bool HasMultipleConstructorsOfSameArity(ref GeneratorExecutionContext con
...
+                if (brackets.Count() > 1) 
                 {
-                    context.ReportDiagnostic(Diagnostic.Create(DiagnosticRules.ArraySignature_JaggedArrayRule, loc, fieldId, typeIdentifier));
-                    return true; // could do or-eq on a `found` boolean instead of exiting as soon as we see invalid...
+                    Report(ref context, DiagnosticRules.ArraySignature_JaggedArrayRule, loc, fieldId, typeIdentifier);
+                    return true;
                 }
-                // [,_] ? 
+                // [,+] ? 
                 else if (brackets.Count() == 1 && brackets.First().ToString().Contains(","))
```
Comment (Scottj1s) [inline src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs:300]: what about white space?

Comment (j0shuams) [reply src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs:300]: Made a test with an array like `[ , , ]` and we raise diagnostic for it, so think this is fine...

**Pull request merged**